### DISCUSSION TITLE
Z offset problem

### DISCUSSION THREAD

**Discussion by ScufaSteve** [Jyers/Marlin#1818] [Q&A]
Hello! I’m new to 3d printing. I’ve spent the last few weeks learning and getting my machine ready for printing TPU drone parts. 

I hope this is the right place to ask this, please redirect me if it isn’t.

Everything has been going pretty smooth until this problem I cannot figure out or find an answer to. 

Current setup:

Ender 3 v2

Micro Swiss direct drive

Dual z axis motors

Antclabs BLTouch

Jyers 2.0.1 custom compiled

Changes from source code (bltouch 3x3):

*machine name changed

*extruder steps calibration 

*X and Y offsets adjusted

*5x5 mesh

Z offset in printer menu is set at -2.49. When commanded to position in this menu it reliably goes back to the perfect nozzle height. I feel like this value is correct.

I run a new mesh and store it to eeprom. All seems normal but the mesh values are incorrect as I find out later…

Start gcode:

; Ender 3 Custom Start G-code

G92 E0 ; Reset Extruder

G28 ; Home all axes

M420 S1

G1 Z2.0 F3000 ; Move Z Axis up little to prevent 
scratching of Heat Bed

G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position

G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line

G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little

G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line

G92 E0 ; Reset Extruder

G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed

G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish

Then it prints with the nozzle too high. Some of the board seems acceptable and it looks like the Z axis is moving just not enough for some of the low spots in the board. 

I have found that if I perform a manual mesh tune and adjust each of the 25 points individually with a feeler gauge and store that mesh to eeprom it prints great. It’s printing a perfect TPU print right now. 

Comparing the auto mesh to the manual mesh they have the same general contour but the low spots on the manual are much deeper. It appears to be running too high in those spots. 

So is my probe inaccurate? Did I miss a setting somewhere? I’ve been trying for figure this out for a few days now, any help would be greatly appreciated!

**Comment by BlueWings172** [comment]
@ScufaSteve 

I noticed that you are running Jyers 2.01. There is an updated version with bug fixes. Check this post #1694 and try this version.

Have you tried UBL ? I use 15x15 UBL which seems pretty accurate. Since the shape of the bed does not change all the time, I only have to probe once every month and only probe 3 points before every print to tilt the bed. Much faster and more accurate than 5x5.

**Reply by ScufaSteve** [reply]
@BlueWings172 

I thought I had the latest release, I will look into that now. 

I haven’t tried UBL yet, looks like that might be the way to go. At least I have an accurate 5x5 mesh to compare the UBL mesh with. I should be able to tell right away if that corrects the issue. 

Thanks!!
